### DISCUSSION TITLE
Branch observation-beta-shared-state-syncups not compiling for watchOS

### DISCUSSION THREAD

**Discussion by Malauch** [pointfreeco/swift-composable-architecture#2718] [General]
### Description

I was checking new *shared state* changes on `observation-beta-shared-state-syncups` branch and noticed that here is a problem which prevents compiling for watchOS: 
https://github.com/pointfreeco/swift-composable-architecture/blob/43f0c25e7fc7185b414d752bec85973b7d682a1c/Sources/ComposableArchitecture/Observation/SharedState/SharedPersistence/FileStorage.swift#L4-L10

UIKit can be imported on watchOS but `UIApplication` is not available. Unfortunately it is not easy fix as adding additional type alias for `WKApplication`. It was added in watchOS 7 and TCA supports watchOS 6. I know that branch is work in progress but decided to share this just in case.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### The Composable Architecture version information

43f0c25e7fc7185b414d752bec85973b7d682a1c

### Destination operating system

watchOS 10

### Xcode version information

Version 15.1 (15C65)

**Comment by mbrandonw** [comment]
Hi @Malauch, that branch is a non-released and non-publicized experimental branch. Some of those findings may never see the light of day. We may get around to fixing the watchOS problem, but no promises. And you are free to open a PR against the branch to fix the problem.

Since this is not an issue with the currently released library, I am going to convert this to a discussion.
